modules/keyless-authentication-robot-accounts.adoc

```diff
@@ -0,0 +1,268 @@
+:_content-type: PROCEDURE
+[id="keyless-authentication-robot-accounts"]
+= Keyless authentication with robot accounts
+
+In previous versions of {productname}, robot account tokens were valid for the lifetime of the token unless deleted or regenerated. Tokens that do not expire have security implications for users who do not want to store long-term passwords or manage the deletion, or regeneration, or new authentication tokens. 
+
+With {productname} {producty}, {productname} administrators are provided the ability to exchange {productname} robot account tokens for an external OIDC token with either Red Hat SSO Key Cloak or Microsoft Entra ID. This allows robot accounts to leverage short-lived, or _ephemeral tokens_, that last one hour. Ephemeral tokens are refreshed regularly and can be used to authenticate individual transactions. 
+
+This feature greatly enhances the security of your {productname} registry by mitigating the possibility of robot token exposure by removing the tokens after one hour.
+
+Configuring keyless authentication with robot accounts is a multi-step procedure that requires setting a robot federation, generating an OAuth2 token from your OIDC provider, and exchanging the OAuth2 token for a robot account access token.
+
+[id="generating-oauth2-token-using-keycloak"]
+== Generating an OAuth2 token with Red Hat SSO Keycloak
+
+The following procedure shows you how to generate an OAuth2 token using Red Hat SSO Keycloak. Depending on your OIDC provider, these steps will vary.
+
+.Procedure
+
+. On the Red Hat SSO Keycloak UI:
+
+.. Click *Clients* and then the name of the application or service that can request authentication of a user.
+
+.. On the *Settings* page of your client, ensure that the following options are set or enabled:
++
+* *Client ID*
+* *Valid redirect URI*
+* *Client authentication*
+* *Authorization*
+* *Standard flow*
+* *Direct access grants*
++
+[NOTE]
+====
+Settings can differ depending on your setup.
+====
+
+.. On the *Credentials* page, store the *Client Secret* for future use.
+
+.. On the *Users* page, click *Add user* and enter a username, for example, `service-account-quaydev`. Then, click *Create*.
+
+.. Click the name of of the user, for example *service-account-quaydev* on the *Users* page. 
+
+.. Click the *Credentials* tab -> *Set password* -> and provide a password for the user. If warranted, you can make this password temporary by selecting the *Temporary* option. 
+
+.. Click the *Realm settings* tab -> *OpenID Endpoint Configuration*. Store the `/protocol/openid-connect/token` endpoint. For example:
++
+[source,text]
+----
+http://localhost:8080/realms/master/protocol/openid-connect/token
+----
+
+. On a web browser, navigate to the following URL:
++
+[source,text]
+----
+http://<keycloak_url>/realms/<realm_name>/protocol/openid-connect/auth?response_type=code&client_id=<client_id>
```

Comment:
The example is not correct, need to add **"auth"**
https://sso-sso.apps.quaytest-3218.qe.azure.devcluster.openshift.com/**auth**/realms/quay/protocol/openid-connect/auth?response_type=code&client_id=quay

Reply:
@LiZhang19817 just to make sure:

I got this URL from OpenID Endpoint Configuration under realm settings. See image for URL. 

Then I navigated to that using the ?response_type=code&client_id=<client_id> metadata. See image for UI endpoint. 

![openid-auth](https://github.com/user-attachments/assets/9b3ce82c-b270-4ff1-8de1-f7e325d8e769)

![code-example](https://github.com/user-attachments/assets/8efbdb01-e6b6-442c-83a2-ad0e6f18bc8c)

In this example, URL is https://localhost:3000/cb?session_state=a81c3488-8eb2-481e-a635-4abb0605303f&iss=http%3A%2F%2Flocalhost%3A8080%2Frealms%2Fmaster&code=0c7dbb1a-584c-4b33-a1f8-9bf90d147051.a81c3488-8eb2-481e-a635-4abb0605303f.cdffafbc-20fb-42b9-b254-866017057f43. The code is what i use with curl -x POST in the following command. 

I'm happy to change to https://sso-sso.apps.quaytest-3218.qe.azure.devcluster.openshift.com/auth/realms/quay/protocol/openid-connect/auth?response_type=code&client_id=quay which might be better? Thanks.